addons/web/static/src/js/views/control_panel/search/control_panel_view_parameters.js

```diff
@@ -24,21 +24,34 @@ var PERIOD_OPTIONS = [
     {description: _lt('Last Year'), optionId: 'last_year', groupId: 3},
 ];
 
+const OPTION_GENERATORS = [
```

Comment:
reorganize this structure into multiple parts, and use the parts in the other files instead of redefining arrays.

Reply:
ok